### DISCUSSION TITLE
AWS.ECS-SERVICE Resize operation

### DISCUSSION THREAD

**Discussion by gabfelp** [cloud-custodian/cloud-custodian#7437] [AWS]
### Ask your question

Hello everyone and thank you very much for the tool.

I'm having a problem using resize on resources AWS.ecs-service. 
The operation is performed (I don't have any errors in the output), **but the change of "desired" does not actually happen** and **no "OffHoursPrevious" tag is generated either.** 
The policy is identical to the one in the [documentation](https://cloudcustodian.io/docs/aws/resources/ecs-service.html#aws-ecs-service-actions-resize):
policies:
  - name: testingecs-service
    resource: aws.ecs-service
    actions:
      - type: resize
        min-capacity: 0
        desired: 0
        save-options-tag: OffHoursPrevious
        suspend-scaling: true

Mysteriously, **when I update desired from the "modify" action it works**:

policies:
  - name: testingecs-service
    resource: aws.ecs-service
    actions:
      - type: modify
        update:
          desiredCount: 0

With this second approach, however, I am unable to save what I had in "desired" earlier. If there is a way to save the old "desired", that way works for me.

What could be the problem? Thank you very much in advance!

### Policy

```shell
policies:
  - name: testingecs-service
    resource: aws.ecs-service
    actions:
      - type: resize
        min-capacity: 0
        desired: 0
        save-options-tag: OffHoursPrevious
        suspend-scaling: true
```

### Relevant log/traceback output

```shell
2022-01-19 19:30:58,176: custodian.policy:INFO policy:testingecs-service resource:aws.ecs-service region:us-east-1 count:1 time:1.96
2022-01-19 19:30:58,758: custodian.policy:INFO policy:testingecs-service action:autoscalingecsservice resources:1 execution_time:0.58
```

**Comment by gabfelp** [comment]
I did a test and print the 'response' variable from line 228 of autoscaling.py

```
for response in response_iterator:
            self.log.info('response %s' % response)           
            for target in response['ScalableTargets']:
                resource_id = target['ResourceId']
```
[...]
Here is the result:
`{'ScalableTargets': [], 'ResponseMetadata': {'RequestId': 'XXXXX', 'HTTPStatusCode': 200, 'HTTPHeaders': {'x-amzn-requestid': XXXX, 'content-type': 'application/x-amz-json-1.1', 'content-length': '22', 'date': 'Thu, 20 Jan 2022 19:27:12 GMT'}, 'RetryAttempts': 0}}`

No 'ScalableTargets' appear in this response, but it doesn't make sense since the policy found a resource earlier, right?
`2022-01-20 16:27:12,151: custodian.policy:INFO policy:testingecs-service resource:aws.ecs-service region:us-east-1 count:1 time:1.92`


Also, this problem only happens on resize. It is still possible to perform the 'modify desiredCount' operation on the filtered ecs-service. Thus, if it were possible to create a tag equal to the current {desiredCount} of the ecs-service resource, that would do it!

Thank you again

**Comment by kapilt** [comment]
resize is targeting the application autoscaling not directly modifying the service, its indirect management so may take some time to settle. ie if your using app autoscaling to control your container service counts, we modify that versus doing a change that the autoscaler will undo. if your not using autoscaling then doing a direct modify can suffice.

**Comment by gabfelp** [comment]
Thanks for the quick response, kapilt, 

Is there any way to keep a tag with the service's previous 'desired' amount? So I could make direct modifications to both "start" and "stop" the service.

Something like:
policies:
  - name: testingecs-service
    resource: aws.ecs-service
    actions:
      - type: modify
        update:
          desiredCount: 0
      - type: tag
        key: previousDesired
        value: '{desiredCount}'
        
for stopping the service, for instance.

Thank you so much!
